game/src/sandbox/dashboards/trip_problems.rs

```diff
@@ -0,0 +1,386 @@
+use std::fmt::Display;
+
+use abstutil::{abbreviated_format, prettyprint_usize};
+use geom::{Angle, Duration, Polygon, Pt2D, Time};
+use map_gui::tools::ColorScale;
+use sim::{Problem, TripMode};
+use widgetry::{Color, DrawWithTooltips, GeomBatch, GeomBatchStack, Text, Widget};
+
+use crate::{App, EventCtx};
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum ProblemType {
+    IntersectionDelay,
+    LargeIntersectionCrossing,
+    OvertakeDesired,
+}
+
+impl ProblemType {
+    pub fn count(self, problems: &Vec<(Time, Problem)>) -> usize {
+        let mut cnt = 0;
+        for (_, problem) in problems {
+            if match problem {
+                Problem::IntersectionDelay(_, _) => self == ProblemType::IntersectionDelay,
+                Problem::LargeIntersectionCrossing(_) => {
+                    self == ProblemType::LargeIntersectionCrossing
+                }
+                Problem::OvertakeDesired(_) => self == ProblemType::OvertakeDesired,
+            } {
+                cnt += 1;
+            }
+        }
+        cnt
+    }
+}
+
+pub trait TripProblemFilter {
```

Comment:
This file was *mostly* just cut/paste/renamed from the risk.rs, but I did make the filter a `trait` so that it could be implemented by both `risk::Filter` and  `travel_times::Filter`.